contracts/ReferralRewards.sol

```diff
@@ -101,13 +101,13 @@ contract ReferalRewards is LPTokenWrapper, IRewardDistributionRecipient {
         emit Staked(msg.sender, amount);
     }
 
-    function stake(uint256 amount, address referral) public updateReward(msg.sender) {
-        require(amount > 0, "Cannot stake 0");
-        super.stake(amount);
-        emit Staked(msg.sender, amount);
+    function stake(uint256 amount, address referral) public {
+        stake(amount);
         
-        // Only set if not set yet
+        // Only set if referral is not set yet
         if(referralOf[msg.sender] == address(0)) {
```

Comment:
Can we instead do:

```
if(referralOf[msg.sender] == address(0) && referral != msg.sender && referral != address(0))
```

So we don't revert but just skip setting the referral?

Reply:
Yes, good call